x-pack/plugins/enterprise_search/public/applications/workplace_search/types.ts

```diff
@@ -96,7 +96,7 @@ export interface ContentSource {
 export interface SourceContentItem {
   id: string;
   last_updated: string;
-  [key: string]: string;
+  [key: string]: string | CustomAPIFieldValue;
```

Comment:
There are several more fields that get returned here, but I decided not to go down the rabbit hole of updating them as it wouldn't fix any bugs,  and would be just a code quality improvement. I added an item to the polish task though.